swift-sdk/Internal/EmbeddedMessagingSerialization.swift

```diff
@@ -4,6 +4,38 @@
 
 import Foundation
 
+struct AnyDecodable: Decodable {
+    let value: Any
+
+    // the current implementation decodes the following value types: null, int, string, bool, double, array, and dictionary
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if container.decodeNil() {
+            value = NSNull()
+        }
+        else if let intVal = try? container.decode(Int.self) {
+            value = intVal
+        } else if let stringVal = try? container.decode(String.self) {
+            value = stringVal
+        } else if let boolVal = try? container.decode(Bool.self) {
+            value = boolVal
+        } else if let doubleVal = try? container.decode(Double.self) {
+            value = doubleVal
+        } else if let arrayVal = try? container.decode([AnyDecodable].self) {
+            value = arrayVal.map { $0.value }
+        } else if let dictionaryVal = try? container.decode([String: AnyDecodable].self) {
+            var dictionary: [String: Any] = [:]
+            for (key, val) in dictionaryVal {
+                dictionary[key] = val.value
+            }
+            value = dictionary
+        } else {
+            throw DecodingError.dataCorruptedError(in: container, debugDescription: "Container contains an unexpected type")
+        }
+    }
+}
+
+
```

Comment:
This is great! But I thought we were just passing along the custom payload as a JSON. But, if this is the route we want to go, we can do something similar in Android.